src/parsing.js

```diff
@@ -102,10 +102,83 @@
     return matchedUnits;
   }
 
+  function ignore(value) {
+    return function(input) {
+      var result = value(input);
+      if (result)
+        result[0] = undefined;
+      return result;
+    }
+  }
+
+  // Regex should be anchored with /^
+  function consumeToken(regex, string) {
+    var result = regex.exec(string);
+    if (result) {
+      result = regex.ignoreCase ? result[0].toLowerCase() : result[0];
+      return [result, string.substr(result.length)];
+    }
+  }
+
+  function consumeNumber(string) {
+    var result = consumeToken(/^[0-9]?\.?[0-9]+/, string);
+    if (result && result[0]) {
```

Comment:
I think I put that there because I ran into a case where there wasn't while I was developing this. However in theory there shouldn't be now that I finished consumeToken (I hope!), so I've removed it :)